posthog/api/sharing.py

```diff
@@ -368,6 +369,55 @@ def retrieve(self, request: Request, *args: Any, **kwargs: Any) -> Any:
                 # We don't want the dashboard to be accidentally loaded via the shared endpoint
                 exported_data.update({"dashboard": dashboard_data})
             exported_data.update({"themes": get_themes_for_team(resource.team)})
+        elif (
+            isinstance(resource, ExportedAsset) and resource.export_context and resource.export_context.get("replay_id")
+        ):
+            # Handle replay export via export_context
+            replay_id = resource.export_context.get("replay_id")
+            timestamp = resource.export_context.get("timestamp")
+
+            if not replay_id:
+                raise NotFound("Invalid replay export - missing replay_id")
+
+            # Create a SessionRecording object for the replay
+            try:
+                # First, try to get existing recording from database
+                try:
+                    recording = SessionRecording.objects.get(session_id=replay_id, team=resource.team)
+                except SessionRecording.DoesNotExist:
+                    # If not found, create it properly
+                    recording = SessionRecording(session_id=replay_id, team=resource.team)
+                    recording.save()  # This ensures it exists in PostgreSQL
+
+                # Now create sharing configuration
+                sharing_config, _ = SharingConfiguration.objects.get_or_create(
+                    team=resource.team,
+                    recording=recording,
+                    defaults={
+                        "enabled": True,
+                        "access_token": secrets.token_urlsafe(32),
+                    },
+                )
+                asset_title = "Session Recording"
+                asset_description = f"Recording {replay_id}"
```

Comment:
talked in person but the more I think about it... i don't think we should be faking a user action to get around a permissions issue and should solve the problem 

if we can do this for an insight / dashboard then we're able to call the query api with an export token, so we should be able to call the snapshots api with an export token (or build a token we can call it with)